insights/parsers/partitions.py

```diff
@@ -0,0 +1,77 @@
+"""
+Partitions - file ``/proc/partitions``
+======================================
+
+This parser reads the ``/proc/partitions`` file, which contains
+partition block allocation information.
+
+"""
+from collections import namedtuple
+
+from insights import Parser, parser, LegacyItemAccess
+from insights.parsers import SkipException, ParseException
+from insights.specs import Specs
+
+
+@parser(Specs.partitions)
+class Partitions(Parser, LegacyItemAccess):
+    """A parser class for parsing the ``/proc/partitions`` file.
+
+    Sample input::
+
+        major minor  #blocks  name
+
+           3     0   19531250 hda
+           3     1     104391 hda1
+           3     2   19422585 hda2
+         253     0   22708224 dm-0
+         253     1     524288 dm-1
+
+    Examples:
+       >>> type(partitions_data)
+       <class 'insights.parsers.partitions.Partitions'>
+       >>> partitions_data[0].name
+       'hda'
+       >>> partitions_data[0].major
+       '3'
+       >>> partitions_data.get_blocks('dm-0')
+       '22708224'
+
+    Raises:
+        SkipException: When input is empty.
+        ParseException: When input cannot be parsed.
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException('Empty content')
+
+        self.data = []
```

Comment:
The `parse_delimited_table` can replace all the following logics.
~~~python
        self.data = parse_delimited_table(content,
                heading_ignore=['major', 'minor'],
                header_substitute=[('#blocks', 'blocks')])
~~~

But you might lose the feature given by the `nametuple`

Reply:
`parse_delimited_table` is cleaner approach and saves lines in the parser. I have modify parser to use `parse_delimited_table` method.
@xiangce  Thanks